src/plugins/intel_gpu/src/graph/impls/cpu/activation.cpp

```diff
@@ -0,0 +1,312 @@
+#include "ngraph/op/ceiling.hpp"
+#include "ngraph/op/erf.hpp"
+#include "ngraph/op/hard_sigmoid.hpp"
+#include "ngraph/op/log.hpp"
+#include "ngraph/op/negative.hpp"
+#include "ngraph/op/selu.hpp"
+#include "ngraph/op/softplus.hpp"
+#include "ngraph/op/tan.hpp"
+#include "ngraph/op/sin.hpp"
+#include "ngraph/op/sinh.hpp"
+#include "ngraph/op/cos.hpp"
+#include "ngraph/op/cosh.hpp"
+#include "ngraph/op/swish.hpp"
+#include "ngraph/op/hswish.hpp"
+#include "ngraph/op/mish.hpp"
+#include "ngraph/op/gelu.hpp"
+#include "ngraph/op/sign.hpp"
+#include "ngraph/op/hsigmoid.hpp"
+#include "ngraph/op/round.hpp"
+
+
+namespace cldnn {
+namespace cpu {
+
+struct activation_impl : public typed_primitive_impl<activation> {
+    using parent = typed_primitive_impl<activation>;
+    using parent::parent;
+
+    activation_func activation_function;
+    activation_additional_params additional_params;
+
+    std::shared_ptr<ov::op::Op> op;
+
+    DECLARE_OBJECT_TYPE_SERIALIZATION
+
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<activation_impl>(*this);
+    }
+
+    activation_impl() : parent("activation_cpu_impl") {}
+
+    explicit activation_impl(const activation_node& outer) {
+        set_node_params(outer);
+    }
+
+    void set_node_params(const program_node& arg) override {
+        IE_ASSERT(arg.is_type<activation>());
+        const auto& node = arg.as<activation>();
+        activation_function = node.get_primitive()->activation_function;
+        additional_params = node.get_primitive()->additional_params;
+    }
+
+    void save(BinaryOutputBuffer& ob) const override {
+        ob << make_data(&activation_function, sizeof(activation_func));
+        ob << make_data(&additional_params, sizeof(activation_additional_params));
+    }
+
+    void load(BinaryInputBuffer& ib) override {
+        ib >> make_data(&activation_function, sizeof(activation_func));
+        ib >> make_data(&additional_params, sizeof(activation_additional_params));
+    }
+
+    template <data_types DT>
+    static void execute_activation(std::shared_ptr<ov::op::Op> op,
+                                   const activation_inst& instance,
+                                   const activation_func& activation_function,
+                                   const activation_additional_params& additional_params) {
+        auto& stream = instance.get_network().get_stream();
+
+        ov::TensorVector input_host_tensors;
+        ov::TensorVector output_host_tensors;
+
+        std::vector<memory::ptr> input_mem_ptrs;
+        for (size_t i = 0; i < instance.dependencies().size(); i++)
+            input_mem_ptrs.push_back(instance.dep_memory_ptr(i));
+
+        // TODO: consider to re-implement lock/unlock in more exception-safetest way
+        for (size_t i = 0; i < input_mem_ptrs.size(); i++)
+            input_host_tensors.push_back(make_tensor(input_mem_ptrs[i]->get_layout(), input_mem_ptrs[i]->lock(stream, mem_lock_type::read)));
```

Comment:
Will it be fair enough usage? In case of mem_lock memory will be unlocked right after this loop - can we consider further memory access as safe in such a case?

Reply:
If you create mem_lock in the loop, then access is valid only within the loop. I've checked the interface of mem_lock, and seems like the usage for multiple inputs will be not that simple (vector of mem_lock unique pointers or something like that). So let's keep it as is for now